packages/@storefront/core/src/services/urlUtils.ts

```diff
@@ -245,7 +245,7 @@ namespace UrlUtils {
 
   export const mergeSearchQueryState = (state: Store.State, request: UrlBeautifier.SearchUrlState) => ({
     ...state.data.present.query,
-    original: request.query || Selectors.query(state),
+    original: request.query === '' ? '' : request.query || Selectors.query(state),
```

Comment:
I would check for `== undefined` instead. Better to check for the fallback case than to check for a special case.